include/seqan3/core/concept/core_language.hpp

```diff
@@ -183,4 +196,17 @@ template <typename t>
 concept standard_layout_concept = std::is_standard_layout_v<t>;
 //!\endcond
 
+/*!\interface   seqan3::weakly_assignable_concept
+ * \brief       Resolves to std::std::is_assignable_v<t>.
+ * \sa          https://en.cppreference.com/w/cpp/types/is_assignable
+ *
+ * \details
+ *
+ * Note that this requires less than std::Assignable!
```

Comment:
Add `It simply tests if the expression std::declval<T>() = std::declval<U>() is well-formed.`